.gitlab-ci.yml

```diff
@@ -1,22 +1,41 @@
 stages:
+  - prepare
   - build
   - deploy-doc
 
 .env:
   before_script:
   - export DOCKER_IMAGE=`echo $CI_PROJECT_NAME | awk '{print tolower($0)}'`:$CI_COMMIT_SHORT_SHA
 
-build_lofar:
+build_base:
+  stage: prepare
+  extends: .env
+  script:
+  - docker build --tag dppp_base:20.04.0 -f ./docker/ubuntu_20_04_base .
```

Comment:
I think you'll need to add a commit hash to this tag. If two different branches of DPPP are building at the same time, they use the same docker registry.

Reply:
So maybe dppp_base_20.04.0:${CI_COMMIT_SHORT_SHA}

Reply:
Agreed, the more so since the base image won't be pushed to a registry. And thus it doesn't hurt if we have a unique (though ugly) tag

Reply:
I did not do that because that tag will also need to be automatically injected into the ubuntu_20_04_lofar file FROM line. I will look into that

Reply:
I think you need something like [this](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact)